### PR TITLE
Yolov2-COCO (13x13x425 output tensor)  #258

### PR DESCRIPTION
**Pull request by leejaymin**
I issued #258. This PR solves #258. 
I completely verified all weights and outputs with the original darknet.
Currently, there is not Yolov2 COCO version. Therefore, I made a new PR.
Thanks.

# YOLOv2-COCO

## Description
This model aims to detect objects in real-time. It detects 80 different classes from the [COCO Datasets](http://cocodataset.org/#home). For information on network architecture, see the [author's page](https://pjreddie.com/darknet/yolov2/) and [white paper](https://arxiv.org/pdf/1612.08242.pdf).

## Model

The model was converted to ONNX from PyTorch version of YOLOv2 using [PyTorch-Yolo2](https://github.com/marvis/pytorch-yolo2). The output is fully verified by generating bounding boxes under PyTorch and onnxruntime.

|Model|Checksum|Download (with sample test data)| ONNX version |Opset version|
|-----|:-------|:-------------------------------|:-------------|:------------|
|YOLOv2|[MD5](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt)|[203.9 MB](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/yolov2-coco/model/yolov2_coco.onnx) |1.5  |9 |

### Input to model
shape `(1x3x416x416)`

### Output of model
shape `(1x425x13x13)`

### Postprocessing steps
The output is a `(1x425x13x13)` tensor where 13x13 is the number of grid cells that the image gets divided into. Each grid cell corresponds to 5 anchors, made up of the 5 bounding boxes predicted by the grid cell and the 80 classes that describe each bounding box (`5 x (80 classes + 5) = 425`). For more information on how to derive the final bounding boxes and their corresponding confidence scores, refer to this [post](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/object-detection-onnx) and [PyTorch source code](https://github.com/marvis/pytorch-yolo2/blob/master/detect.py).

## Dataset (Train and validation)
The YOLOv2 model was trained on the [COCO](http://cocodataset.org/#home) datasets and was sourced from the original yolov2-voc `.cfg` and `.weights` files from [link](https://pjreddie.com/darknet/yolov2/).

## References
"YOLO9000: Better, Faster, Stronger" [arXiv:1612.08242](https://arxiv.org/pdf/1612.08242.pdf)

## License
MIT License

### CODE REVIEW

File: vision/object_detection_segmentation/yolov2-coco/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# YOLOv2-COCO
+
```
Comment (vinitra) [inline vision/object_detection_segmentation/yolov2-coco/README.md]: The .gitattributes file in this folder is not required -- please delete.

File: .gitattributes [hunk 2]
```diff
@@ -6,3 +6,4 @@
 *.pkl filter=lfs diff=lfs merge=lfs -text
 *.tar.gz filter=lfs diff=lfs merge=lfs -text
 *.tgz filter=lfs diff=lfs merge=lfs -text
+vision/object_detection_segmentation/yolov2-coco/model/yolov2_coco.onnx filter=lfs diff=lfs merge=lfs -text
```
Comment (vinitra) [inline .gitattributes]: It's not necessary to explicitly encode this file in .gitattributes.

File: vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+MD5 (yolov2_coco.onnx) = dd60e6fb27a5956b7705c985e1f62737
```
Comment (vinitra) [inline vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt]: Can you format this as:
dd60e6fb27a5956b7705c985e1f62737 yolov2_coco.onnx

Comment (vinitra) [inline vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt]: Additionally, MD5 file can be stored in the models folder (and the link on the README accordingly updated!) We're trying to standardize the storage of these files across all models.

File: vision/object_detection_segmentation/yolov2-coco/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+## Description
+This model aims to detect objects in real time. It detects 80 different classes from the [COCO Datasets](http://cocodataset.org/#home). For information on network architecture, see the [author's page](https://pjreddie.com/darknet/yolov2/) and [white paper](https://arxiv.org/pdf/1612.08242.pdf).
+
+## Model
+
```
Comment (vinitra) [inline vision/object_detection_segmentation/yolov2-coco/README.md]: nit: extra whitespace

File: vision/object_detection_segmentation/yolov2-coco/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# YOLOv2-COCO
```
Comment (vinitra) [inline vision/object_detection_segmentation/yolov2-coco/README.md]: Please also include a link to this folder from the README in the source of this repository and describe the difference between this offering and the other YOLOv2 model link in the description.

File: vision/object_detection_segmentation/yolov2-coco/README.md (hunk 1 above)
Comment (leejaymin) [reply vision/object_detection_segmentation/yolov2-coco/README.md]: Thank you for your comment. I deleted it.

File: .gitattributes (hunk 2 above)
Comment (leejaymin) [reply .gitattributes]: Thank you for your comment. I deleted it.

File: vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt (hunk 3 above)
Comment (leejaymin) [reply vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt]: Thank you for your comment. I changed it

File: vision/object_detection_segmentation/yolov2-coco/README.md (hunk 4 above)
Comment (leejaymin) [reply vision/object_detection_segmentation/yolov2-coco/README.md]: Thank you for your comment. I changed it

File: vision/object_detection_segmentation/yolov2-coco/README.md (hunk 5 above)
Comment (leejaymin) [reply vision/object_detection_segmentation/yolov2-coco/README.md]: Thank you for your comment. I revised README to add a link to the yolov2-coco folder.

File: vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt (hunk 3 above)
Comment (leejaymin) [reply vision/object_detection_segmentation/yolov2-coco/yolov2_coco_md5.txt]: Thank you for your comment. I moved it.

File: vision/object_detection_segmentation/yolov2-coco/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+shape `(1x3x416x416)`
+
+### Output of model
+shape `(1x425x13x13)`
+
+### Postprocessing steps
+The output is a `(1x425x13x13)` tensor where 13x13 is the number of grid cells that the image gets divided into. Each grid cell corresponds to 5 anchors, made up of the 5 bounding boxes predicted by the grid cell and the 80 classes that describe each bounding box (`5 x (80 classes + 5) = 425`). For more information on how to derive the final bounding boxes and their corresponding confidence scores, refer to this [post](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/object-detection-onnx) and [PyTorch source code](https://github.com/marvis/pytorch-yolo2/blob/master/detect.py).
+
+## Dataset (Train and validation)
+The YOLOv2 model was trained on the [COCO](http://cocodataset.org/#home) datasets and was sourced from the original yolov2-voc `.cfg` and `.weights` files from [link](https://pjreddie.com/darknet/yolov2/).
```
Comment (iranroman) [inline vision/object_detection_segmentation/yolov2-coco/README.md:24]: This is a typo, right? It should read:

from the original yolov2-coco .cfg and .weights files from ...

but the yolov2-coco file does not exist anyway. Can you please clarify how this onnx file was generated? Thank you!

Comment (iranroman) [reply vision/object_detection_segmentation/yolov2-coco/README.md:24]: Or do you mean that you took the `cfg` from the original yolov2-voc and you retrained the model on the COCO dataset? 

This is very confusing. I've been trying to use the yolov2-coco-9.onnx model and it shows some very erratic behavior. All other Yolo V2 models in this repo behave normally. 

Please clarify. Thanks.

Comment (iranroman) [reply vision/object_detection_segmentation/yolov2-coco/README.md:24]: I looked into the potential darknet source code for this model, and the issue is that this `onnx` file assumes that the input image is divided by `255` before entering the network.

**Pull request merged**